### PR TITLE
listener: Don't error w/ 500 for superfluous events

### PR DESCRIPTION
**Pull request by oxzi**
When, for example, resubmitting a state change, the listener will error with an internal server error, code 500. This is both a bit heavy and misleading.

The error is now inspected similar to the current behavior for the icinga2.Launcher in its launch method.

### CODE REVIEW

File: internal/listener/listener.go [hunk 1]
```diff
@@ -133,7 +134,16 @@ func (l *Listener) ProcessEvent(w http.ResponseWriter, req *http.Request) {
...
 	l.logger.Infow("Processing event", zap.String("event", ev.String()))
 	err = incident.ProcessEvent(context.Background(), l.db, l.logs, l.runtimeConfig, &ev)
-	if err != nil {
+	switch {
+	case errors.Is(err, event.ErrSuperfluousStateChange):
+		abort(http.StatusAlreadyReported, &ev, "Stopped processing event with superfluous state change: %v", err)
+		return
+
+	case errors.Is(err, event.ErrSuperfluousMuteUnmuteEvent):
+		abort(http.StatusAlreadyReported, &ev, "Stopped processing event with superfluous (un)mute object: %v", err)
```
Comment (julianbrost) [inline internal/listener/listener.go:143]: The messages given here are somewhat redundant with what's already part of `err.Error()`, so this results in responses like this:

```
Stopped processing event with superfluous state change: ignoring superfluous state change: ok state event from source 1
```

**Review by julianbrost — CHANGES_REQUESTED** [review]

File: internal/listener/listener.go (hunk 1 above)
Comment (oxzi) [reply internal/listener/listener.go]: I have unified both cases and picked another <small>arbitrary</small> HTTP status code. Now the returned message looks like:
> ignoring superfluous state change: ok state event from source 2

File: internal/listener/listener.go [hunk 2]
```diff
@@ -133,7 +134,14 @@ func (l *Listener) ProcessEvent(w http.ResponseWriter, req *http.Request) {
...
 	err = incident.ProcessEvent(context.Background(), l.db, l.logs, l.runtimeConfig, &ev)
-	if err != nil {
+	switch {
+	case errors.Is(err, event.ErrSuperfluousStateChange):
+		fallthrough
+	case errors.Is(err, event.ErrSuperfluousMuteUnmuteEvent):
+		abort(http.StatusNotAcceptable, &ev, "%v", err)
+		return
+
+	case err != nil:
```
Comment (julianbrost) [inline internal/listener/listener.go:144]: I already noticed that you're a way bigger fan of `switch` than I am, but with the `fallthrough` here, a regular `if` looks way nicer to me:
```suggestion
	if errors.Is(err, event.ErrSuperfluousStateChange) || errors.Is(err, event.ErrSuperfluousMuteUnmuteEvent) {
		abort(http.StatusNotAcceptable, &ev, "%v", err)
		return
	} else if err != nil {
```

Comment (oxzi) [reply internal/listener/listener.go]: Fair enough.

**Review by julianbrost — APPROVED** [review]

**Pull request merged**